backend/src/controllers/MeetingController.ts

```diff
@@ -0,0 +1,166 @@
+import { Request, Response } from 'express';
+import asyncHandler from 'express-async-handler';
+import Meeting, { MeetingModel } from '../models/MeetingModel';
+import Attendance from '../models/AttendanceModel';
+import { MeetingRequest } from '../types/RequestTypes';
+import { TypedRequestBody, TypedRequest } from '../types/UtilTypes';
+import { MeetingIdParam } from '../types/RequestParams';
+import PaginationHandler from '../classes/PaginationHandler';
+import { MeetingFilterQuery } from '../types/QueryTypes';
+
+/**
+ * @desc 	Get all the meetings
+ * @route 	GET /api/meetings/
+ */
+const getAllMeetings = asyncHandler(
+    new PaginationHandler<MeetingModel, MeetingFilterQuery>(Meeting)
+        .pre((query, req) => {
+            const filterHandlers: Record<string, (value: string) => void> = {
+                before: (value) =>
+                    (query = query.where({ time: { $lt: new Date(value) } })),
+                after: (value) =>
+                    (query = query.where({ time: { $gt: new Date(value) } })),
+                time: (value) =>
+                    (query = query.where('time').equals(new Date(value))),
+                creator: (value) =>
+                    (query = query.where('creator').equals(value)),
+                location: (value) =>
+                    (query = query.where('location', new RegExp(value, 'i'))),
+                name: (value) =>
+                    (query = query.where('name', new RegExp(value, 'i'))),
+            };
+
+            Object.entries(req.query)
+                .filter(([query]) => filterHandlers[query])
+                .forEach(([query, value]: [string, string]) =>
+                    filterHandlers[query](value),
+                );
+
+            if (!req.query.passed || req.query.passed !== 'true') {
+                query = query.where('time').gte(Date.now()); // Only get events that haven't passed
+            }
+
+            return query;
+        })
+        .pre((query) => query.sort({ time: 'ascending' }))
+        .post(
+            async (events) =>
+                await Promise.all(
+                    events.map((event) => event.populate('creator')),
+                ),
+        ).handler,
+);
+
+/**
+ * @desc    Get a specific meeting
+ * @route   GET /api/meetings/:meetingId
+ */
+const getMeeting = asyncHandler(
+    async (req: Request<MeetingIdParam>, res: Response) => {
+        const meeting = await Meeting.findById(req.params.meetingId);
+        if (meeting) {
+            await meeting.populate('creator');
+            res.status(200).json({
+                status: 'success',
```

Comment:
> A few small changes that I missed on the first pass and I was wondering if we should move the `invited` field from the `MeetingModel` to the `AttendanceModel`?

Yeah, I think that would make sense. It would be easier to keep track of the attendance and would make more sense contextually. I'll edit the models to adapt to this.

Also, the `attendance` object is being populated :)